lib/urlForCargoPackage.nix

```
@@ -19,8 +19,10 @@ let
     else if nameLen == 3 then "3/${substr 0 1 name}"
     else "${substr 0 2 name}/${substr 2 4 name}";
 
+  knownRegistries = lib.concatStringsSep " " (builtins.attrNames crateRegistries);
   registry = crateRegistries.${source} or (throw ''
     not sure how to download crates from ${source}.
+    known registries are ${knownRegistries}.
```

```suggestion
  knownRegistries = lib.concatStringsSep "\n" (builtins.attrNames crateRegistries);
  registry = crateRegistries.${source} or (throw ''
    not sure how to download crates from ${source}.
    known registries are:
    ${knownRegistries}
    
```

Might be better to print these out on one line each for easier scanning